main.go

```diff
@@ -1,132 +1,336 @@
 package main
 
 import (
+	"bytes"
+	"flag"
 	"fmt"
-	"strings"
+	"log"
+	"net/http"
+	_ "net/http/pprof"
+	"os"
+	"os/signal"
+	"runtime"
+	"sync/atomic"
+	"syscall"
 	"time"
 
-	"github.com/valyala/fasthttp"
-	"github.com/vkcom/kittenhouse/core/clickhouse"
-	"github.com/vkcom/kittenhouse/core/cmd"
-	"github.com/vkcom/kittenhouse/core/inmem"
-	"github.com/vkcom/kittenhouse/core/persist"
+	"github.com/vkcom/kittenhouse/clickhouse"
+	"github.com/vkcom/kittenhouse/destination"
+	"github.com/vkcom/kittenhouse/inmem"
+	"github.com/vkcom/kittenhouse/persist"
+	"github.com/vkcom/kittenhouse/config"
+	"github.com/vkcom/engine-go/srvfunc"
+)
+
+var (
+	// Build* заполняются при сборке go build -ldflags
+	BuildTime    string
+	BuildOSUname string
+	BuildCommit  string
+	buildVersion string // объединение Build* в одну строку
+)
+
+var (
+	argv struct {
+		reverse bool
+
+		disableKittenMeow bool
+
+		protocol   string
+		host       string
+		port       uint
+		help       bool
+		version    bool
+		markAsDone bool
+		user       string
+		group      string
+		log        string
+		logDebug   bool
+
+		maxOpenFiles      uint64
+		nProc             uint
+		pprofHostPort     string
+		chHost            string
+		config            string
+		dir               string
+		maxBufferSize     int64
+		maxSendSize       int64
+		maxFileSize       int64
+		rotateIntervalSec int64
+	}
+
+	logFd *os.File
 )
 
 const (
-	defaultTimeout = 30 * time.Second
+	FlagPersistent = 1 << 0
+	FlagDebug      = 1 << 1
+	FlagRowBinary  = 1 << 2
 )
 
+
+
+const (
+	heartbeatInterval = time.Hour
+	maxUDPPacketSize  = 8192 // does not make sense to make it much more than that
+)
+
+func init() {
+	buildVersion = fmt.Sprintf(`kittenhouse compiled at %s by %s after %s on %s`, BuildTime, runtime.Version(),
+		BuildCommit, BuildOSUname,
+	)
+
+	log.SetFlags(log.LstdFlags | log.Lshortfile | log.Lmicroseconds)
+
+	// actions
+	flag.BoolVar(&argv.help, `h`, false, `show this help`)
+	flag.BoolVar(&argv.version, `version`, false, `show version`)
+	flag.BoolVar(&argv.reverse, `reverse`, false, `start reverse proxy server instead (ch-addr is used as clickhouse host-port)`)
+
+	// common options
+	flag.StringVar(&argv.protocol, `protocol`, `rpc`, `protocol to use; rpc or http`)
```

Comment:
This is a breaking change. You need to try to launch kittenhouse with the same parameters as before (in the open-source version) and make sure it can even start.

Reply:
done